tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py

```diff
@@ -187,6 +195,91 @@ def test_backward_gpu_im2col_nobias(self):
                             None, cuda.to_gpu(self.gy),
                             use_cudnn='never')
 
+    def check_double_backward(self, x_data, W_data, b_data, y_grad,
+                              x_grad_grad, W_grad_grad, b_grad_grad,
+                              use_cudnn='always'):
+        xp = cuda.get_array_module(x_data)
+
+        if not self.c_contiguous:
+            x_data = xp.asfortranarray(x_data)
+            W_data = xp.asfortranarray(W_data)
+            y_grad = xp.asfortranarray(y_grad)
+            x_grad_grad = xp.asfortranarray(x_grad_grad)
+            W_grad_grad = xp.asfortranarray(W_grad_grad)
+            self.assertFalse(x_data.flags.c_contiguous)
+            self.assertFalse(W_data.flags.c_contiguous)
+            self.assertFalse(y_grad.flags.c_contiguous)
+            self.assertFalse(x_grad_grad.flags.c_contiguous)
+            self.assertFalse(W_grad_grad.flags.c_contiguous)
+            if b_data is not None:
+                b = xp.empty((len(b_data) * 2,), dtype=self.b.dtype)
+                b[::2] = b_data
+                b_data = b[::2]
+                self.assertFalse(b_data.flags.c_contiguous)
+
+                ggb = xp.empty((len(b_data) * 2,), dtype=self.b.dtype)
+                ggb[::2] = b_grad_grad
+                b_grad_grad = ggb[::2]
+                self.assertFalse(b_grad_grad.flags.c_contiguous)
+
+        args = (x_data, W_data)
+        grad_grads = (x_grad_grad, W_grad_grad)
+        if b_data is not None:
+            args = args + (b_data,)
+            grad_grads = grad_grads + (b_grad_grad,)
+
+        def f(*args):
+            y = F.convolution_nd(*args, stride=self.stride, pad=self.pad,
+                                 cover_all=self.cover_all)
+            return y * y  # make the function nonlinear
+
+        with chainer.using_config('use_cudnn', use_cudnn):
+            with chainer.using_config('autotune', self.autotune):
+                gradient_check.check_double_backward(
+                    f, args, y_grad, grad_grads,
+                    dtype='d', atol=5e-3, rtol=5e-2)
+
+    @condition.retry(3)
```

Comment:
Could you make the tests pass without `retry` by adjusting the tolerances? In general, I think we want to avoid relying on `retry` since it may hide bugs or numerical instabilities.

Reply:
Is there any discussions on introduction of `condition.retry` decorator in GitHub? I think it is useful and important in the sense that is avoids the problem that the test fails stochastically.

Reply:
Not as I am aware of unfortunately. Do you know how frequently this test fails for a reasonable atol, rtol? It would make sense if a function contains non-differentiable points but else, we shouldn't need it (slightly off topic but it's addressed in this PR https://github.com/chainer/chainer/pull/3499)? But again, I see it's aligned with the `Convolution2D` tests, so it's fine I think given low/small margin failures.

Reply:
In my environment it fails with the probability of 10%. Of course it can be solved by carefully adjusting the parameters of atol and rtol, however, I think relaxation of such parameters is essentially equivalent to the use of `condition.retry`. At least the tests of `Convolution2D` should also be changed for consistency if we want to remove `condition.retry` decorators.

Reply:
They're not really equivalent in the sense that a function containing non-differentiable points may be detected by adjusting the tolerances but are hidden with retries. 

But I see that e.g. Convolution2D tests have the same problem (also see https://github.com/chainer/chainer/issues/3986). I am currently running the Convolution2D tests with 1000 repetitions per fixture. So far ~80% done and they've all passed. I discussed with @niboshi about the failure rate and with this in mind the rate seems a bit high. Is there anything we can do about the current failure rate?

Sorry for the really long reply!!

Reply:
> They're not really equivalent in the sense that a function containing non-differentiable points may be detected by adjusting the tolerances but are hidden with retries.
> Is there anything we can do about the current failure rate?
? Sorry, but I'm not sure why you regard this problem as important. Do you want to detect a non-differentiable point that `Convolution2D` has?
Honestly speaking, I'm not sure why it is so important, but I'm OK to fix this issue if you consider it is so important, but I would be glad if you would tell me the best parameters of `rtol` and `atol`.